### ISSUE TITLE
Add examples of more complex NpyIter usage to the documentation.

### ISSUE THREAD

**Issue by ewmoore** [numpy/numpy#5695]
The iterator is nice in some regards, simple things are simple, and simply iterating over all elements is straightforward.  However, how to do things like what `PyArray_IterAllButAxis` does but with multiple broadcasted arrays, etc is pretty opaque. (To me at least; maybe I'm just dense.)

My currently go at using nested iterators to try things out is [here](https://gist.github.com/ewmoore/e88ee9dd84c1d9d58892), FWIW.  I'm performing a reduction along a specified tuple of axes, computing both the sum and product stored in the same output array.  Its a bit of mess, but it does seem to work. Doing this kind of thing appears to take quite a lot of setup though.

**Comment by jaimefrio** [comment]
Yes indeed it does... I have an example in #4330 of nested iteration, and the amount of code needed is ridiculous. I have said elsewhere that `NpyIter` needs a middle layer that makes it usable in a more general setting, and `PyArray_IterAllButAxis` is a great example of something that is now orders of magnitude harder to put together.

**Comment by shiningsurya** [comment]
_sorry to revive a really old issue_

i have to do some `IterAllButAxis` on multiple `numpy.ndarray` and having to setup nested iterators on all of them seems counter intuitive. 

Are there any plans to extend `NpyIter` ?

is it okay to use `PyArray_IterAllButAxis` until an alternative is available?
